src/player.js

```diff
@@ -482,6 +484,35 @@ export default class Player extends FakeEventTarget {
     }
   }
 
+  /**
+   * Checks for callbacks that should change the tracks, and call them on the
+   * respective track group (audio/text/video)
+   * @private
+   * @returns {void}
+   */
+  _maybeSetTracksLabels() {
+    const labelsCallbacks = this._config.customLabels;
+    if (labelsCallbacks) {
+      for (let callbackType in labelsCallbacks) {
+        this._setTracksCustomLabels(this._getTracksByType(LabelToTrackMap[callbackType]), labelsCallbacks[callbackType]);
+      }
+    }
+  }
+
+  /**
+   *
+   * @param {Array<Track>} tracks - tracks
+   * @param {Function} callback - application label callback, returns a string
+   * @private
+   * @returns {void}
+   */
+  _setTracksCustomLabels(tracks: Array<Track>, callback: Function) {
+    tracks.forEach(track => {
+      const callbackResult = callback(Utils.Object.copyDeep(track));
+      track.label = typeof callbackResult === 'string' ? callbackResult : track.label;
```

Comment:
no need in re-assignment for the label field in case the callback returns no answer
```
if (typeof result === 'string') {
   track.label = result;
}
```

Reply:
done